x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -1225,6 +1226,10 @@ private Expression resolveConvertFunction(AbstractConvertFunction convert, List<
             if (convert.field() instanceof FieldAttribute fa && fa.field() instanceof InvalidMappedField imf) {
                 HashMap<TypeResolutionKey, Expression> typeResolutions = new HashMap<>();
                 Set<DataType> supportedTypes = convert.supportedTypes();
+                if (convert instanceof FoldablesConvertFunction fcf && supportedTypes.isEmpty()) {
```

Comment:
Isn't the `instanceof` check sufficient, like so?
```suggestion
                if (convert instanceof FoldablesConvertFunction fcf) {
```

Reply:
Yes, `instanceof` is sufficient, right now all of `FoldablesConvertFunction` have empty `supportedTypes`, a little too cautious here, I'll remove it.